src/main/java/com/gooddata/http/client/GoodDataHttpClient.java

```diff
@@ -84,301 +205,177 @@ private GoodDataChallengeType identifyGoodDataChallenge(final ClassicHttpRespons
     }
 
     /**
-     * Helper method to clone request entity safely, handling both repeatable and non-repeatable entities.
-     * For non-repeatable entities, buffers the content to allow reuse.
+     * Refresh temporary token.
+     *
+     * @return <ul>
+     * <li><code>true</code> TT refresh successful</li>
+     * <li><code>false</code> TT refresh unsuccessful (SST expired)</li>
+     * </ul>
+     * @throws GoodDataAuthException error
      */
-    private <T extends ClassicHttpRequest> T cloneRequestWithEntity(T target, ClassicHttpRequest source) throws IOException {
-        if (!(source instanceof org.apache.hc.core5.http.HttpEntityContainer)) {
-            return target;
-        }
-        
-        org.apache.hc.core5.http.HttpEntity entity = 
-            ((org.apache.hc.core5.http.HttpEntityContainer) source).getEntity();
-        
-        if (entity == null) {
-            return target;
-        }
-        
-        // Check if entity is repeatable - if so, we can reuse it directly
-        if (entity.isRepeatable()) {
-            if (target instanceof org.apache.hc.core5.http.HttpEntityContainer) {
-                ((org.apache.hc.core5.http.HttpEntityContainer) target).setEntity(entity);
-            }
-        } else {
-            // Entity is not repeatable - buffer it for reuse
-            log.debug("Buffering non-repeatable entity for retry");
-            byte[] content = EntityUtils.toByteArray(entity);
-            String contentTypeStr = entity.getContentType();
-            ContentType contentType = contentTypeStr != null ? 
-                ContentType.parseLenient(contentTypeStr) : ContentType.DEFAULT_BINARY;
-            
-            ByteArrayEntity bufferedEntity = new ByteArrayEntity(content, contentType);
-            if (target instanceof org.apache.hc.core5.http.HttpEntityContainer) {
-                ((org.apache.hc.core5.http.HttpEntityContainer) target).setEntity(bufferedEntity);
-            }
-        }
-        
-        return target;
-    }
-
     private boolean refreshTt() throws IOException {
         log.debug("Obtaining TT");
+
         final HttpGet request = new HttpGet(TOKEN_URL);
         try {
-
-            request.addHeader(SST_HEADER, sst);
-
-            return httpClient.execute(authHost, request, (HttpContext) null, response -> {
-                int status = response.getCode();
-
-                switch (status) {
-                    case HttpStatus.SC_OK:
-                        tt = TokenUtils.extractTT(response);
-                        return true;
-                    case HttpStatus.SC_UNAUTHORIZED:
-                        return false;
-                    default:
-                        throw new GoodDataAuthException("Unable to obtain TT, HTTP status: " + status);
+            request.setHeader(SST_HEADER, sst);
+            final HttpResponse response = httpClient.execute(authHost, request, (HttpContext) null);
```

Comment:
It's strongly recommended to ensure that the response object gets eventually closed e.g. by enclosing the respective block in a try-with-resources statement.
```suggestion
            try (ClassicHttpResponse response = httpClient.execute(authHost, request, (HttpContext) null)) {
```